### ISSUE TITLE
[youtube] Skipping player response / HTTP Error 429: Too Many Requests / Video Not Available

### ISSUE THREAD

**Issue by userlip** [yt-dlp/yt-dlp#7143]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.03.04** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Germany

### Provide a description that is worded well enough to be understood

Downloading any video currently gives me the error of "No Video formats found" @coletdjnz already said it seems to be related to the API giving 429 responses. Here is my verbose log:

```
yt-dlp "https://www.youtube.com/shorts/y7aFRsM7vRU" -v
[debug] Command-line config: ['https://www.youtube.com/shorts/y7aFRsM7vRU', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d]
[debug] Lazy loading extractors is disabled
[debug] Python 3.8.10 (CPython x86_64 64bit) - Linux-5.15.0-56-generic-x86_64-with-glibc2.29 (OpenSSL 1.1.1f  31 Mar 2020, glibc 2.31)
[debug] exe versions: ffmpeg n4.4-179-gcb0a1bb805-20211011 (fdk,setts), ffprobe n4.4-179-gcb0a1bb805-20211011
[debug] Optional libraries: Cryptodome-3.18.0, brotli-1.0.9, certifi-2023.05.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-11.0.3
[debug] Proxy map: {}
[debug] Loaded 1822 extractors
[youtube] Extracting URL: https://www.youtube.com/shorts/y7aFRsM7vRU
[youtube] y7aFRsM7vRU: Downloading webpage
WARNING: [youtube] Unable to download webpage: HTTP Error 429: Too Many Requests
[youtube] y7aFRsM7vRU: Downloading android player API JSON
WARNING: [youtube] Skipping player response from android client (got player response for video "aQvGIIdgFDM" instead of "y7aFRsM7vRU"); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
[youtube] y7aFRsM7vRU: Downloading iframe API JS
[youtube] y7aFRsM7vRU: Downloading player bbe1b497
[youtube] y7aFRsM7vRU: Downloading web player API JSON
WARNING: [youtube] Skipping player response from web client (got player response for video "aQvGIIdgFDM" instead of "y7aFRsM7vRU"); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
[youtube] y7aFRsM7vRU: Downloading initial data API JSON
WARNING: Extractor failed to obtain "title". Creating a generic title instead
ERROR: [youtube] y7aFRsM7vRU: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/yt_dlp/YoutubeDL.py", line 1552, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/yt_dlp/YoutubeDL.py", line 1649, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/usr/local/lib/python3.8/dist-packages/yt_dlp/YoutubeDL.py", line 1708, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/usr/local/lib/python3.8/dist-packages/yt_dlp/YoutubeDL.py", line 2650, in process_video_result
    self.raise_no_formats(info_dict)
  File "/usr/local/lib/python3.8/dist-packages/yt_dlp/YoutubeDL.py", line 1066, in raise_no_formats    raise ExtractorError(msg, video_id=info['id'], ie=info['extractor'],
yt_dlp.utils.ExtractorError: [youtube] y7aFRsM7vRU: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
```

I made sure to update to the latest version using this command:

```
sudo python3 -m pip install --force-reinstall git+https://github.com/yt-dlp/yt-dlp.git --no-cache-dir --upgrade
```

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d]
[debug] Lazy loading extractors is disabled
[debug] Python 3.8.10 (CPython x86_64 64bit) - Linux-5.15.0-56-generic-x86_64-with-glibc2.29 (OpenSSL 1.1.1f  31 Mar 2020, glibc 2.31)
[debug] exe versions: ffmpeg n4.4-179-gcb0a1bb805-20211011 (fdk,setts), ffprobe n4.4-179-gcb0a1bb805-20211011
[debug] Optional libraries: Cryptodome-3.18.0, brotli-1.0.9, certifi-2023.05.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-11.0.3
[debug] Proxy map: {}
[debug] Loaded 1822 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
yt-dlp is up to date (stable@2023.03.04)
```

**Comment by Dibbyo456** [comment]
I believe your ip was throttled, trying with proxy should work.

**Comment by unixfox** [comment]
Related invidious issue: https://github.com/iv-org/invidious/issues/3822

@userlip is it possible to tell us the provider of your server?

**Comment by userlip** [comment]
@unixfox I use a Hetzner Cloud Server

**Comment by unixfox** [comment]
Would you be able to send me the IPv4 address of that server? You can send it to me on matrix (`@unixfox:matrix.org`) or by email (see my github profile).

I'm trying to correlate this issue because on invidious side we had this issue only for people on hetzner (invidious.vpsburti.com and invidious.nerdvpn.de).

**Comment by userlip** [comment]
Really sorry I cant help you with the exact IP as this would break a contractual agreement I have but I can confirm to you that it only happens on Hetzner servers. I have 2 more instances running as a backup on other providers and they are doing fine.

**Comment by M4hbod** [comment]
I'm having the same issue on both of my Hetzner servers.

**Comment by unixfox** [comment]
> I'm having the same issue on both of my Hetzner servers.

Like my request in https://github.com/yt-dlp/yt-dlp/issues/7143#issuecomment-1565627856, would you be able to share these IPs :)?

(Btw I'm not a random, I'm a maintainer at @iv-org)

**Comment by gamer191** [comment]
after some debugging with @userlip, we've confirmed this affects the official youtube website, at least for shorts
The regular website (which pages specifically @userlip?) redirects to a recaptcha

**Comment by fermani10** [comment]
I'm having the same problem on my Hetzner server, does anyone know the solution?

**Comment by gamer191** [comment]
do `--force-ipv4` or `--force-ipv6` fix this? The invidious issue suggests the latter will

**Comment by userlip** [comment]
--force-ipv4 actually works! ipv6 does not!

**Comment by gamer191** [comment]
can you please post verbose logs of each

**Comment by dikmen0fficial** [comment]
I'm having the same problem on my Hetzner server, does anyone know the solution?


:(

**Comment by gamer191** [comment]
I already posted a workaround 🤦:


> do `--force-ipv4` or `--force-ipv6` fix this? The invidious issue suggests the latter will

**Comment by dikmen0fficial** [comment]
> Zaten bir geçici çözüm gönderdim🤦:
> 
> > bunu yap `--force-ipv4`veya `--force-ipv6`düzelt? Hakaret konusu, ikincisinin olacağını gösteriyor

How can I fix this? can you help with **anydesk** about this please?

**Comment by dikmen0fficial** [comment]
> I already posted a workaround 🤦:
> 
> > do `--force-ipv4` or `--force-ipv6` fix this? The invidious issue suggests the latter will

How can I fix this? can you help with **anydesk** about this please?

**Comment by Ulmo** [comment]
Similar issue.  Does this also happen with Digital Ocean servers?  I tried both --force-ipv4 and --force-ipv6 and neither worked.  I'll have to consider doing a proxy.

**Comment by unixfox** [comment]
> Similar issue. Does this also happen with Digital Ocean servers? I tried both --force-ipv4 and --force-ipv6 and neither worked. I'll have to consider doing a proxy.

Are you encountering this issue on Digitalocean too? If so, which region?

Or your question is if digital ocean is affected too? Are you currently on hetzner?

**Comment by M4hbod** [comment]
> One invidious maintainer told me that he can't reproduce the issue anymore since he shutted down their invidious instance since last Saterday.
> 
> The people that reported this issue when using yt-dlp, can you still replicate the issue?
> 
> @Ulmo @userlip @dikmen0fficial @M4hbod

No

**Comment by huntercoke** [comment]
My server is at Hetzner but I am using a proxy and still facing the issue.

**Comment by fermani10** [comment]
I also continue with the problem, temporarily mount my telegram bot on another server. But I would like to continue using Hetzner. Idk why this happens...

**Comment by unixfox** [comment]
For the people that have IPv6 on their server, could you try if changing the IPv6 address in the same range as your server solves the issue?

You can add more IPv6 addresses manually using: https://tldp.org/HOWTO/Linux+IPv6-HOWTO/ch06s02.html
Or changing the network settings of your server.

There is also an advanced way to automatically "rotate" your IPv6 address with this software here: https://github.com/ycd/ipv6-rotator

**Comment by cc00010** [comment]
@unixfox If I don't have ipv6 and can't use proxy because it's so slow? any other options?

**Comment by pukkandan** [comment]
The blocking seems to be purely IP based and server side. Since even browsers do not seem to be immune, there is no solution that doesn't involve changing your IP somehow (passing logged in cookies may also help). There is really nothing we can do about this.

**Comment by huntercoke** [comment]
@pukkandan it started to work for me, I changed nothing. It seems like a soft block, I was using a proxy and cookie.

**Comment by artshevtsov** [comment]
--force-ipv6 worked for me, I am using Hetzner Cloud

**Comment by rdfsx** [comment]
I faced the same issue on hetzner

**Comment by fermani10** [comment]
I was able to download like this  yt-dlp https://www.youtube.com/watch?v=qXSYp4IDdNs --source-address MYIP

How do I implement this in my bot?

    ydl_opts = {
        "format": "best",
        "outtmpl": f"{video['id']}.%(ext)s",
    } ?

**Comment by bashonly** [comment]
```diff
 ydl_opts = {
     "format": "best",
     "outtmpl": f"{video['id']}.%(ext)s",
+    "source_address": "0.0.0.0",
 }
```

**Comment by buddhhu** [comment]
Same issue but only for live streams. I am trying to get m3u8 link by using -g and it's giving 429 error

**Comment by schmorp** [comment]
Unfortunately, my bug reoport was closed as a duplicate of this one, but I don't think it is:

yt-dlp should not claim the download succeeded when it is obviously and easily-testably blocked. Instead, yt-dlp should fail and not mark the video as a successful download - this is clearly a bug in yt-dlp, especially as it is trivial to test for this case.

**Comment by gamer191** [comment]
@schmorp that's a separate issue which is being tracked at #7715. `-f "[ext!=mhtml]"` should work around it

**Comment by fermani10** [comment]
![image](https://github.com/yt-dlp/yt-dlp/assets/31140793/a238a8c3-b2e8-476c-b60b-977948a6b0e5)

I still have problems, I tried rotating IP and everything and the blocking continues. I don't know if it's the server or what the problem is. There must be some solution.

```
Test
        ydl_opts = {
            'format': 'bestvideo[ext!=mhtml]',
            'outtmpl': '%(title)s - %(extractor)s-%(id)s.%(ext)s',
            'source_address': 'ip-server',
            'writethumbnail': True
        }
        
test        
                ydl_opts = {
       'format': 'best[ext=mp4]',
            'outtmpl': '%(title)s - %(extractor)s-%(id)s.%(ext)s',
            #'source_address': 'ip-server',
            'writethumbnail': True
        }
```

**Comment by testrustic** [comment]
> ![image](https://private-user-images.githubusercontent.com/31140793/292331946-a238a8c3-b2e8-476c-b60b-977948a6b0e5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxOTg5ODQsIm5iZiI6MTcwMzE5ODY4NCwicGF0aCI6Ii8zMTE0MDc5My8yOTIzMzE5NDYtYTIzOGE4YzMtYjJlOC00NzZjLWI2MGItOTc3OTQ4YTZiMGU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIxVDIyNDQ0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNmMGEyNjRmZDA2YTdiMDQ0NTZhYTg3MWJjYjRkZjc3MDgyNWMzM2Y0YzQ4MDU2YzJiOTI4NWE3NzdiNzczM2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UYdRHiRNhzGtP6ayR8__18s5Inlw10WwI9xiSywmD-A)
> 
> I still have problems, I tried rotating IP and everything and the blocking continues. I don't know if it's the server or what the problem is. There must be some solution.
> 
> ```
> Test
>         ydl_opts = {
>             'format': 'bestvideo[ext!=mhtml]',
>             'outtmpl': '%(title)s - %(extractor)s-%(id)s.%(ext)s',
>             'source_address': 'ip-server',
>             'writethumbnail': True
>         }
>         
> test        
>                 ydl_opts = {
>        'format': 'best[ext=mp4]',
>             'outtmpl': '%(title)s - %(extractor)s-%(id)s.%(ext)s',
>             #'source_address': 'ip-server',
>             'writethumbnail': True
>         }
> ```

If you tried forcing and changing the IP. You should try changing servers or your code. Although it seems like a permanent ban from YouTube. but it's strange that they ban all your IP's

**Comment by FacuSeb** [comment]
With cookies and youtube premium it is solved at the moment it is slower but it downloads

**Comment by juliawilkins** [comment]
I am facing a similar issue, getting the following error: 
```
WARNING: [youtube] Skipping player response from ios client (got player response for video "M5t4UHllkUM" instead of "-1Hub6Ps_cc"); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
WARNING: Only images are available for download. use --list-formats to see them
ERROR: [youtube] -1Hub6Ps_cc: Requested format is not available. Use --list-formats for a list of available formats <dest path>
```

I've never gotten the error thrown for too many requests, so I'm not positive that my IP address is blocked or not. Any tips?

**Comment by bashonly** [comment]
@juliawilkins try passing the `-4` or `-6` options to force ipv4 or ipv6, or else try passing logged-in youtube cookies to yt-dlp

**Comment by juliawilkins** [comment]
Thanks @bashonly, I had tried passing ipv4/ipv6 but no luck. Have you tried the youtube cookies route? I'm a little confused on how to actually implement that. Thanks!

**Comment by bashonly** [comment]
@juliawilkins if you are running this on a server then youtube has certainly blocked your server's IP (if not a whole range). You'll need to export cookies from a browser following [the workaround steps here](https://github.com/yt-dlp/yt-dlp/issues/8227#issuecomment-1867184804) or else use a proxy / different IP

**Comment by juliawilkins** [comment]
Thanks a lot @bashonly will give this a shot.

**Comment by LollenApe** [comment]
Hi.
I'm having the same kind of error with Paperspace, by DigitalOcean.
Since I'm not using it for server stuff but just to make python scripts/jupyter notebooks (using Paperspace "Gradient", which is only notebook-based), I can't install extensions or whatever.
Already tried -4/-6 and -f "[ext!=mhtml]" but none of these worked, if anyone could find a solution, thanks in advance.

**Comment by unixfox** [comment]
The Invidious team has developed a tool for rotating the IPv6 address over an IPv6 range: https://github.com/iv-org/smart-ipv6-rotator

For the moment this escape easily the blockage from YouTube and has been the solution for all the Invidious instances.

**Comment by LollenApe** [comment]
Cool but I don't think I can find my IPv6 subnet, or even calculate it with that cgi file...

**Comment by LollenApe** [comment]
Cool, but I don't think I can get IPv6 address from Paperspace... Tried everything, lmao.

**Comment by adnenabdessaied** [comment]
> Thanks a lot @bashonly will give this a shot.
@juliawilkins Could you please confirm if the proposed solution worked for you and/or how you managed to solve the issue? That would be very appreciated

**Comment by adnenabdessaied** [comment]
@juliawilkins  Could you please confirm if the proposed solution worked for you and/or how you managed to solve the issue? That would be very appreciated

**Comment by Ali-Flt** [comment]
Any solutions for hetzner (Helsinki) servers? I tried cookies, forcing ipv4 and ipv6 but nothing works. I can't download live videos and it only happens on the hetzner server itself. Interestingly, if I use yt-dlp on another machine with a proxy to the hetzner server it works without any issues. But running yt-dlp on the hetzner server itself I encounter HTTP Error 429 warnings and also live videos won't download.

**Comment by philinp** [comment]
Hetzner Germany 94.130.58.*:
> WARNING: [youtube+oauth2] Failed to download m3u8 information: HTTP Error 429: Too Many Requests
-4/-6 don't help.
any suggestions?

**Comment by bashonly** [comment]
These errors don't really happen anymore.

If an android client request is made with bad player params, this is the new result:

```
[youtube] ___: Downloading android player API JSON
ERROR: [youtube] ___: The following content is not available on this app.. Watch on the latest version of YouTube.
```

and we have since removed the hardcoded player params from the extractor.

And when a DC IP has been blocked, the new error given is one of these two:
- https://github.com/yt-dlp/yt-dlp/issues/10128
- https://github.com/yt-dlp/yt-dlp/issues/10085

Closing this issue as obsolete.
